diff -Nur old/usr/lib/python3/test/support/__init__.py new/usr/lib/python3/test/support/__init__.py
--- old/usr/lib/python3/test/support/__init__.py	2018-01-04 19:30:12.113594629 +0100
+++ new/usr/lib/python3/test/support/__init__.py	2018-01-04 19:30:46.209413191 +0100
@@ -675,17 +675,20 @@

 def _is_ipv6_enabled():
     """Check whether IPv6 is enabled on this host."""
-    if socket.has_ipv6:
-        sock = None
-        try:
-            sock = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
-            sock.bind((HOSTv6, 0))
-            return True
-        except OSError:
-            pass
-        finally:
-            if sock:
-                sock.close()
+    try:
+        if socket.has_ipv6:
+            sock = None
+            try:
+                sock = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
+                sock.bind((HOSTv6, 0))
+                return True
+            except OSError:
+                pass
+            finally:
+                if sock:
+                    sock.close()
+    except Exception:
+        pass
     return False

 IPV6_ENABLED = _is_ipv6_enabled()
